lib/reek/spec/should_reek_of.rb

```diff
@@ -18,6 +18,7 @@ def initialize(smell_type_or_class,
         @smell_details = smell_details
         @configuration = configuration
         @configuration.load_values(smell_type => { Smells::SmellConfiguration::ENABLED_KEY => true })
+        @matching_smell_types = nil
```

Comment:
>> Adding the ivar assignment fix this class' ivar assumption.

Now here's the problem: I'd consider initialising instance variables to nil as not fixing anything. 
There is no semantic difference between:

```Ruby
class Omg
  def initialize; @bar = nil; end
  def hola(arg); @bar = arg; end
end
```

and

```Ruby
class Omg
  def hola(arg); @bar = arg; end
end
```

Hence, it should still be a smell. 
@mvz @chastell wdyt?

I think initialising instance variables to nil is a strong hint that.....this shouldn't be an instance variable at all.
I'm fine with leaving the instance variable here to not block this pull request (after all, it was there before) but I think we need to discuss my question above since it is relevant for the detector.